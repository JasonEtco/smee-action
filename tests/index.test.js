jest.mock('@actions/core')

const core = require('@actions/core')
const nock = require('nock')
const run = require('../entrypoint')

describe('smee-action', () => {
  let data

  beforeEach(() => {
    nock('https://api.github.com')
      .get(/\/repos\/.*\/.*/).reply(200, {
        id: 123
      })

    nock('https://smee.io')
      .post(/.*/).reply(200, (match, body) => {
        data = { match, body }
      })
  })

  it('makes a POST request to smee.io using the repository\'s id', async () => {
    await run()

    expect(data.match).toBe('/123')
    expect(data.body).toMatchSnapshot()
  })

  it('makes a POST request to smee.io using smeeChannel input', async () => {
    core.getInput = jest
      .fn()
      .mockReturnValueOnce('pizzadog')

    await run()

    expect(data.match).toBe('/pizzadog')
    expect(data.body).toMatchSnapshot()
  })

  it('exits with a failure if the POST fails', async () => {
    nock.cleanAll()
    nock('https://smee.io').post(/.*/).replyWithError(500)
    nock('https://api.github.com')
      .get(/\/repos\/.*\/.*/).reply(200, { id: 123 })

    await run()

    expect(core.setFailed).toHaveBeenCalled()
    expect(core.setFailed.mock.calls[0][0]).toMatchSnapshot()
  })
})
