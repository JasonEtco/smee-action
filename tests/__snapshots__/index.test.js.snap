// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smee-action exits with a failure if the POST fails 1`] = `"request to https://smee.io/123 failed, reason: 500"`;

exports[`smee-action makes a POST request to smee.io using smeeChannel input 1`] = `
Object {
  "repository": Object {
    "name": "in-a-coffee-shop",
    "owner": Object {
      "login": "JasonEtco",
    },
  },
  "smee-action": Object {
    "action": "GITHUB_ACTION",
    "actor": "GITHUB_ACTOR",
    "event": "GITHUB_EVENT_NAME",
    "ref": "GITHUB_REF",
    "sha": "GITHUB_SHA",
    "workflow": "GITHUB_WORKFLOW",
  },
}
`;

exports[`smee-action makes a POST request to smee.io using the repository's id 1`] = `
Object {
  "repository": Object {
    "name": "in-a-coffee-shop",
    "owner": Object {
      "login": "JasonEtco",
    },
  },
  "smee-action": Object {
    "action": "GITHUB_ACTION",
    "actor": "GITHUB_ACTOR",
    "event": "GITHUB_EVENT_NAME",
    "ref": "GITHUB_REF",
    "sha": "GITHUB_SHA",
    "workflow": "GITHUB_WORKFLOW",
  },
}
`;
